app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java

```diff
@@ -1297,7 +1297,7 @@ private Mono<Application> checkoutRemoteBranch(String defaultApplicationId, Stri
                             .flatMap(application1 -> addAnalyticsForGitOperation(
                                     AnalyticsEvents.GIT_CHECKOUT_REMOTE_BRANCH.getEventName(),
                                     application1,
-                                    application1.getGitApplicationMetadata().getIsRepoPrivate()
+                                    Optional.ofNullable(application1.getGitApplicationMetadata().getIsRepoPrivate()).isPresent() && application1.getGitApplicationMetadata().getIsRepoPrivate()
```

Comment:
I think this can be simplified to:

```suggestion
                                    Boolean.TRUE.equals(application1.getGitApplicationMetadata().getIsRepoPrivate())
```

If value is `true`, we get `true`, if `false`, we get `false`, if `null`, we get `false`, which is the same behavior.